File: elixir/04-mad_lib/mad_lib.ex
Code.load_file("../util/input.ex")

defmodule MadLib do
  import Input

  def run do
    noun = string("Enter a noun: ")
    verb = string("Enter a verb: ")
    adjective = string("Enter an adjective: ")
    adverb = string("Enter an adverb: ")

    print(noun, verb, adjective, adverb)
  end

  def print(noun, verb, adjective, adverb) do
    IO.puts("Do you #{verb} with your #{adjective} #{noun} #{adverb}? That's hilarious!")
  end

  # TODO implement a branching story, where the answers to questions determine
  # how the story is constructed. (Hint: use Structs to build the branchs)

end

MadLib.run()

File: kotlin/04-mad_lib/MadLib.kt
import Input.string

fun main(args: Array<String>) {
    val noun = string("Enter a noun: ")
    val verb = string("Enter a verb: ")
    val adjective = string("Enter an adjective: ")
    val adverb = string("Enter an adverb: ")

    println("Do you $verb your $adjective $noun $adverb? That's hilarious!")
}
